modules/reactor/src/meshgenerators/HexagonMeshTrimmer.C

```diff
@@ -0,0 +1,566 @@
+          boundary_info.add_side((*elem_it)->neighbor_ptr(i),
+                                 ((*elem_it)->neighbor_ptr(i))->which_neighbor_am_i(*elem_it),
+                                 trimming_section_boundary_id);
+        }
+      }
+      mesh.delete_elem(*elem_it);
+    }
+  }
+  mesh.contract();
+  mesh.prepare_for_use();
+
+  // Assign customized outer surface boundary id and name
+  if (external_boundary_id != OUTER_SIDESET_ID)
+    MooseMesh::changeBoundaryId(mesh, OUTER_SIDESET_ID, external_boundary_id, false);
+}
+
+bool
+HexagonMeshTrimmer::lineSideDeterminator(const Real px,
+                                         const Real py,
+                                         const Real param_1,
+                                         const Real param_2,
+                                         const Real param_3,
+                                         const bool direction_param,
+                                         const Real dis_tol)
+{
+  const Real tmp = px * param_1 + py * param_2 + param_3;
+  return direction_param ? tmp >= dis_tol : tmp <= dis_tol;
+}
+
+Point
+HexagonMeshTrimmer::twoLineIntersection(const Real param_11,
+                                        const Real param_12,
+                                        const Real param_13,
+                                        const Real param_21,
+                                        const Real param_22,
+                                        const Real param_23)
+{
+  return Point(
+      (param_12 * param_23 - param_22 * param_13) / (param_11 * param_22 - param_21 * param_12),
+      (param_13 * param_21 - param_23 * param_11) / (param_11 * param_22 - param_21 * param_12),
+      0.0);
+}
+
+void
+HexagonMeshTrimmer::quasiTriElementsFixer(ReplicatedMesh & mesh,
+                                          const std::set<subdomain_id_type> subdomain_ids_set)
+{
+  BoundaryInfo & boundary_info = mesh.get_boundary_info();
+  const subdomain_id_type max_subdomain_id(*subdomain_ids_set.rbegin());
+  const subdomain_id_type tri_subdomain_id_shift =
+      _tri_elem_subdomain_shift == Moose::INVALID_BLOCK_ID ? max_subdomain_id
+                                                           : _tri_elem_subdomain_shift;
+  const unsigned int n_elem_extra_ids = mesh.n_elem_integers();
+  std::vector<dof_id_type> exist_extra_ids(n_elem_extra_ids);
+  std::vector<std::tuple<Elem *, unsigned int, bool>> bad_elems_rec;
+  unsigned int bad_elem_counter = 0;
+  for (auto & elem : as_range(mesh.active_elements_begin(), mesh.active_elements_end()))
+  {
+    auto elem_angles = vertex_angles(*elem);
+    if (MooseUtils::absoluteFuzzyEqual(elem_angles.front().first, M_PI, 0.001))
+    {
+      bad_elems_rec.push_back(std::make_tuple(elem, elem_angles.front().second, false));
+      bad_elem_counter++;
+    }
+  }
+  unsigned int bad_elem_ext = 0;
+  std::set<subdomain_id_type> new_subdomain_ids;
+  for (unsigned i = 0; i < bad_elem_counter; i++)
```

Comment:
i is only used to access the element right? so you could use a range-based iterator?